src/tree/containerizedFunctionApp/ResolvedContainerizedFunctionAppResourceBase.ts

```diff
@@ -0,0 +1,150 @@
+import { envUtils } from "../../utils/envUtils";
+import { treeUtils } from "../../utils/treeUtils";
+import { type ApplicationSettings, type FuncHostRequest } from "../IProjectTreeItem";
+import { ProjectSource } from "../projectContextValues";
+import { ContainerAppSettingsClientProvider } from "./AppSettingsClient";
+import { type ContainerTreeItem } from "./ContainerTreeItem";
+import { FunctionsTreeItem } from "./FunctionsTreeItem";
+import { ImageTreeItem } from "./ImageTreeItem";
+
+export class ResolvedContainerizedFunctionAppResource implements ResolvedAppResourceBase {
+    public site: Site;
+    public maskedValuesToAdd: string[] = [];
+
+    public appSettingsTreeItem: AppSettingsTreeItem;
+    private _functionsTreeItem: FunctionsTreeItem;
+    private _imageTreeItem: ImageTreeItem;
+
+    private _cachedVersion: FuncVersion | undefined;
+    private _cachedHostJson: IParsedHostJson | undefined;
+
+    public readonly source: ProjectSource = ProjectSource.Remote;
+
+    public constructor(site: Site) {
+        this.site = site;
+    }
+
+    public static async createResolvedFunctionAppResource(context: IActionContext, subscription: ISubscriptionContext, site: Site): Promise<ResolvedContainerizedFunctionAppResource> {
+        const resource = new ResolvedContainerizedFunctionAppResource(site);
+        const client = await createWebSiteClient([context, subscription]);
+        resource.site.siteConfig = await client.webApps.getConfiguration(nonNullProp(resource.site, 'resourceGroup'), nonNullProp(resource.site, 'name'));
+        return resource;
+    }
+
+    public get name(): string {
+        return this.label;
+    }
+
+    public get label(): string {
+        return nonNullProp(this.site, 'name');
+    }
+
+    public get id(): string {
+        return nonNullProp(this.site, 'id');
+    }
+
+    public get iconPath(): TreeItemIconPath {
+        return treeUtils.getIconPath('azFuncProductionSlot');
+    }
+
+    public get viewProperties(): ViewPropertiesModel {
+        return {
+            data: this.site,
+            label: this.name,
+        }
+    }
+
+    public hasMoreChildrenImpl(): boolean {
+        return false;
+    }
+
+    public async loadMoreChildrenImpl(_clearCache: boolean): Promise<AzExtTreeItem[]> {
+        const proxyTree: ContainerTreeItem = this as unknown as ContainerTreeItem;
+
+        this._functionsTreeItem = new FunctionsTreeItem(proxyTree, this.site);
+        this._imageTreeItem = new ImageTreeItem(proxyTree, this.site, this.maskedValuesToAdd);
+        this.appSettingsTreeItem = new AppSettingsTreeItem(proxyTree, new ContainerAppSettingsClientProvider(proxyTree, proxyTree.subscription), ext.prefix, {
+            contextValuesToAdd: ['azFunc']
+        });
+
+        const children: AzExtTreeItem[] = [this._functionsTreeItem, this._imageTreeItem, this.appSettingsTreeItem]
+
+        return children;
+    }
+
+    public async isReadOnly(context: IActionContext): Promise<boolean> {
```

Comment:
You probably don't need to implement this the same way. I assume that this can just be true.